gfsphysics/GFS_layer/GFS_typedefs.F90

```diff
@@ -3340,6 +3330,15 @@ subroutine control_initialize (Model, nlunit, fn_nml, me, master, &
     Model%ialb             = ialb
     Model%iems             = iems
     Model%iaer             = iaer
+    if (iaer/1000 == 1 .or. Model%iccn == 2) then
+      Model%iaerclm = .true.
+      ntrcaer = ntrcaerm
+    else
+      ntrcaer = 1
+    endif
+#ifdef CCPP
+    Model%ntrcaer          = ntrcaer
```

Comment:
Is this truly specific to CCPP?

Reply:
Yes, it is. This has got to do with the fact that in CCPP arrays need to be allocated before the physics are initialized (i.e. before the climatology is initialized and the local `ntrcaer` variable in `aerclm_def` is set correctly). In IPD, initializing the arrays and the physics are interwoven, hence the variable `ntrcaer` in `GFS_typedefs.F90` is only required for CCPP. I can explain that in more detail to you in the next week if you think this is useful.